File: interviewbit-bit-manipulation-min-xor-value.py
class Solution:
    # @param A : list of integers
    # @return an integer
    def findMinXor(self, A):
        A.sort()
        min_diff = float("inf")
        for i in range(len(A) - 1):
            min_diff = min(min_diff, A[i + 1] ^ A[i])

        return min_diff

File: interviewbit-bit-manipulation-min-xor-value.java
import java.util.Collections;

public class Solution {
    public int findMinXor(ArrayList<Integer> A) {
        if (A.size() < 2) {
            return 0;
        }
        Collections.sort(A);
        int minXor = A.get(0) ^ A.get(1);
        for (int i = 1; i < A.size() - 1; i++) {
            minXor = Math.min(minXor, A.get(i) ^ A.get(i + 1));
        }
        return minXor;
    }
}
